### PR TITLE
breakdown of Catalyst using rollup

### PR DESCRIPTION
**Pull request by rcowin**

### CODE REVIEW

File: catalyst/src/catalyst.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+            target[name] = initializeSandbox(name); 
+            rv = Reflect.get(target, name, receiver);
+            renderRule.sandboxes.push(rv);
+        }
+        rv.lastAccessTime = new Date().getTime();
+        return rv;
+    }
+  });
+
+  const spaCleanupThreshold = 500;
```
Comment (briannorman) [inline catalyst/src/catalyst.js:20]: Occasionally, I think this threshold still causes me issues.  I don't recall exactly, but is this a performance thing or why not just consider every access fresh?

File: catalyst/src/enode.js [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+ENode.prototype.filter = function(sel){
+  var el = this.el;
+  if (!sel) return this;
+  return new ENode(el.filter(function(e){
+    return e.matches(sel)
+  }))
+};
+ENode.prototype.contains = function (text) {
+  var el = this.el;
+  return new ENode(el.filter(function (e) { return e.textContent.includes(text)}));
```
Comment (briannorman) [inline catalyst/src/enode.js:43]: I would change `return e.textContent.includes(text)` to `return e.textContent.indexOf(text) > -1` as IE11 doesn't support `includes`.  See https://caniuse.com/?search=includes.  Even though we exclude IE from our experiments, the context still runs for VBG and they often complain of errors they see on IE11.

File: catalyst/src/intervalHandler.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+      this.clearIntervalTimer();
+
+      function process() {
+        var baseItems = this.triggerQueue.slice();
+        var items = baseItems;
+
+        var results;
+        do {
+          items = results || items;
+          results = items.filter(function(task){
```
Comment (briannorman) [inline catalyst/src/intervalHandler.js:44]: IE sucks and doesn't support `filter`.  https://caniuse.com/?search=filter

**Review by briannorman — APPROVED** [review]
There were a few IE things that are probably worth cleaning up but, overall, looks good to me!

File: catalyst/src/catalyst.js (hunk 1 above)
Comment (rcowin) [reply catalyst/src/catalyst.js:20]: There is a mechanism to provide your own check or to disable it. 
This is to compensate for the fact that we don't have a mechanism to query the eid or context id that is active for the code executing. If we did, we could have be an explicit and clean mechanism.

As an example, my build mechanism for vscode adds the following line to the bottom of the context code:
```
    rule.isActive = function(){
      return [...document.querySelector('html').classList].includes("evolv_web_${context.id}")
    };
```

This works very well for all circumstances, but requires the context code to know it's context id in the yaml.

**Pull request merged**